ncm-network/src/main/perl/nmstate.pm

```diff
@@ -330,10 +355,11 @@ sub generate_nmstate_config
             $ifaceconfig->{state} = "up";
         }
     } elsif ($is_vlan_eth) {
-        my $vlan_id = $name;
-        # replace everything up-to and including . to get vlan id of the interface.
-        # TODO: instead of this, should perhaps add valid-id in schema? but may not be backward compatible for existing host entreis, aqdb will need updating?
-        $vlan_id =~ s/^[^.]*.//;;
+        my $vlan_id = $self->find_vlan_id($name, $iface->{device});
+        # if vlan_id is empty, error
+        if (! $vlan_id) {
+            $self->error("Could not find vlan id for vlan device $name");
```

Comment:
shouldn't you return here? or skip the rest of the code?

Reply:
erm not sure actually,  This will still create the config, with no vlan id:  , will it will be empty and therefore nmstate apply will also fail, which will be more clear as what has gone wrong?

Reply:
if you are sure it will fail, it's ok for me